src/conventional/amdar_bufr2ioda.py

```diff
@@ -0,0 +1,508 @@
+        return np.array(data, dtype=object)
+    else:
+        if data.dtype == float:
+            data[np.abs(data) >= np.abs(bufr_missing_value)] = float_missing_value
+            return np.array(data, dtype=np.float32)
+        elif data.dtype == int:
+            data[np.abs(data) >= np.abs(bufr_missing_value)] = int_missing_value
+            return np.array(data, dtype=np.int32)
+        elif data.dtype.kind in {'U', 'S'}:
+            #data[data == ""] = string_missing_value
+            return np.array(data, dtype=object)
+    logging.critical("ABORT, no matching datatype found for key: " + key)
+
+
+# Assign the correct IODA missing value to MetaData element.
+def assign_missing_meta(key, mimic):
+
+    if (locationKeyList[meta_keys.index(key)][1] == "string"):
+        return np.full(len(mimic), string_missing_value, dtype=object)
+    elif (locationKeyList[meta_keys.index(key)][1] == "integer"):
+        return np.full(len(mimic), int_missing_value, dtype=np.int32)
+    elif (locationKeyList[meta_keys.index(key)][1] == "long"):
+        return np.full(len(mimic), long_missing_value, dtype=np.int64)
+    elif (locationKeyList[meta_keys.index(key)][1] == "float"):
+        return np.full(len(mimic), float_missing_value, dtype=np.float32)
+    elif (locationKeyList[meta_keys.index(key)][1] == "double"):
+        return np.full(len(mimic), double_missing_value, dtype=np.float64)
+    else:
+        logging.critical("ABORT, no matching datatype found for key: " + key)
+
+
+# Return True if all elements in the array are missing, otherwise False.
+def is_all_missing(data):
+    if data.dtype == np.float32:
+        return all(x==float_missing_value for x in data)
+    if data.dtype == np.float64:
+        return all(x==double_missing_value for x in data)
+    elif data.dtype == np.int32:
+        return all(x==int_missing_value for x in data)
+    elif data.dtype == np.int64:
+        return all(x==long_missing_value for x in data)
+    elif data.dtype.kind in {'U', 'S'}:
+        return all(x==string_missing_value for x in data)
+
+    logging.warning("data type not determined, returning False")
+    return False
+
+
+def read_file(file_name, count, start_pos, data):
+
+    f = open(file_name, 'rb')
+
+    while True:
+        # Use eccodes to decode each bufr message in the file
+        data, count, start_pos = read_bufr_message(f, count, start_pos, data)
+
+        if start_pos is None:
+            break
+
+    return data, count, start_pos
+
+
+def read_bufr_message(f, count, start_pos, data):
+
+    meta_data = {}         # All the various MetaData go in here.
+    vals = {}              # Floating-point ObsValues variables go in here.
+    avals = []             # Temporarily hold array of values.
+    call_fail = False
+    if start_pos == f.tell():
+        return data, count, None
+    start_pos = f.tell()
+    count[0] += 1
+
+    met_utils = meteo_utils.meteo_utils()
+
+    try:
+        bufr = codes_bufr_new_from_file(f)
```

Comment:
I assume this is one example where this *should* be `eccodes.codes_bufr_new_from_file(f)`.